src/PoolManager.sol

```diff
@@ -314,34 +300,10 @@ contract PoolManager is Auth, IPoolManager {
     }
 
     /// @inheritdoc IPoolManager
-    function updateMember(uint64 poolId, bytes16 trancheId, address user, uint64 validUntil) public auth {
-        require(user != address(escrow), "PoolManager/escrow-member-cannot-be-updated");
-
-        TrancheTokenLike trancheToken = TrancheTokenLike(getTrancheToken(poolId, trancheId));
-        require(address(trancheToken) != address(0), "PoolManager/unknown-token");
-
-        RestrictionManagerLike restrictionManager = RestrictionManagerLike(address(trancheToken.restrictionManager()));
-        restrictionManager.updateMember(user, validUntil);
-    }
-
-    /// @inheritdoc IPoolManager
-    function freeze(uint64 poolId, bytes16 trancheId, address user) public auth {
-        require(user != address(escrow), "PoolManager/escrow-cannot-be-frozen");
-
+    function updateRestriction(uint64 poolId, bytes16 trancheId, bytes memory update) public auth {
```

Comment:
Changed to an `UpdateRestriction` message to be more generic, and more easily allow adding custom restriction types for future hooks.